packages/relay/src/lib/model.ts

```diff
@@ -122,4 +122,48 @@ export class Receipt {
         this.logsBloom = '';
         this.status = record.receipt.status == Status.Success ? "0x1" : "0x0";
     }
+}
+
+export class Transaction {
+    public readonly accessList!: string | null;
+    public readonly blockHash!: string | null;
+    public readonly blockNumber!: string | null;
+    public readonly chainId!: string;
+    public readonly from!: string;
+    public readonly gas!: string;
+    public readonly gasPrice!: string;
+    public readonly hash!: string;
+    public readonly input!: string;
+    public readonly maxPriorityFeePerGas!: string;
+    public readonly maxFeePerGas!: string;
+    public readonly nonce!: string;
+    public readonly r!: string;
+    public readonly s!: string;
+    public readonly to!: string | null;
+    public readonly transactionIndex!: string | null;
+    public readonly type!: string;
+    public readonly v!: string;
+    public readonly value!: string;
+
+    constructor(args: any) {
```

Comment:
For the functions that sonar complains about too-many-params, should we adopt this pattern? @Nana-EC @mmrt

Reply:
Yes. In my PR I created a few interfaces so we're passing expandable objects instead of individual params.
Also had this class in my PR but Ivo gets to get it in first :)